django/db/models/base.py

```diff
@@ -2455,6 +2456,29 @@ def _check_long_column_names(cls, databases):
 
         return errors
 
+    @classmethod
+    def _check_related_fields(cls):
+        has_db_variant = False
+        has_python_variant = False
+        for rel in cls._meta.get_fields():
+            if rel.related_model and not rel.auto_created:
+                if not (on_delete := getattr(rel.remote_field, "on_delete", None)):
+                    continue
+                if isinstance(on_delete, DatabaseOnDelete):
+                    has_db_variant = True
+                else:
```

Comment:
Same problem here, suspect we want something like:
```suggestion
                elif on_delete != DO_NOTHING:
```